### PR TITLE
add location model tests

### PR DESCRIPTION
**Pull request by willblake01**
Leads to more stable application by adding test coverage.

-Add attachment traits to location factory
-Add class_name to events association
-Add spec/support/assets/ directory
-Add location model tests

### CODE REVIEW

**Review by runbeyond — APPROVED** [review]

File: app/models/location.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def should_generate_new_friendly_id?
     slug.blank? || name_changed?
   end
 
-  has_many :events
+  has_many :events, class_name: "::Event"
```
Comment (ColinW520) [inline app/models/location.rb:8]: really shouldn't need to specify this... were you running into something here?

File: spec/models/location_spec.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      expect(subject.class.reflect_on_association(:races).macro).to eq(:has_many)
+    end
+
+    it "should have one cover_photo attached association" do
+      expect(subject.class.reflect_on_association(:cover_photo_attachment).macro).to eq(:has_one)
+    end
+
+    it "should have many promo_photos attached association" do
+      expect(subject.class.reflect_on_association(:promo_photos_attachments).macro).to eq(:has_many)
+    end
```
Comment (ColinW520) [inline spec/models/location_spec.rb:23]: This is wayyyyy overkill, but, probably a good thing to learn. Let's talk about where to draw the line on 'does this test make sense?'

Comment (willblake01) [reply spec/models/location_spec.rb:23]: Cool. Yeah, I was down a model testing rabbit hole and unsure of where that line was.

File: app/models/location.rb (hunk 1 above)
Comment (willblake01) [reply app/models/location.rb:8]: I was getting an ActiveStorage error and found that as a fix, which worked in the moment. Now that I removed it the tests are still passing so maybe something else I subsequently changed resolved the same issue.

**Pull request closed without merging**